frontend/type_test.py

```diff
@@ -0,0 +1,24 @@
+from heir import compile
+from heir.mlir import I64, Secret
+
+from absl.testing import absltest  # fmt: skip
+
+
+class TypeTest(absltest.TestCase):
+
+  def test_type(self):
+
+    @compile()
+    def type_two(x: Secret[I64], y: Secret[I64]):
```

Comment:
I must be missing something, because it seems like this test doesn't actually involve any type extension?

Reply:
Constants need to be type extended - the constant 2 is initially an i32, and is the LHS SSA value for the `2 * x` - this was the regression test. 
(In a testcase we need a non block arg, e.g. a constant, to be mismatched with the input types)

Reply:
added a comment here explaining

Reply:
Oh, I'm blind, completely missed the constant `2` in there.